src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSnapshotManagerDispatcherBase.cs

```diff
@@ -68,9 +78,10 @@ private void ThreadStart()
                         var task = _tasks.Take();
                         TryExecuteTask(task);
                     }
-                    catch (ThreadAbortException)
+                    catch (ThreadAbortException ex)
                     {
-                        // Fires when things shut down or in tests. Swallow thread abort exceptions and bail out.
+                        // Fires when things shut down or in tests. Log exception and bail out.
+                        _logException(ex);
```

Comment:
Surprisingly I tested and it doesn't seem to get hit during shutdown. It does seem to get hit during tests though

Reply:
This seems scary to me, that we just stop the dispatcher for any exception. Should we exclude `OperationCancelledException` for instance? Or does `TryExecuteTask` do its own, better, error handling?

Reply:
That's a fair point, I guess technically for all non-thread aborted exceptions we could log and just continue executing

Reply:
Hmm, good point. I wonder if we could just go back to handling `ThreadAbortException`, since that's the main exception we're concerned about here? The other exceptions would just throw which seems acceptable to me since that's the current behavior today.

Reply:
Actually your right, they'd throw, so we're still be stopping the dispatcher. Maybe I'm worried about nothing.

Reply:
They'd throw as is but this is an opportunity to pick up on unexpected scenarios when they throw and optimistically continue executing 😄

Reply:
Yeah, thinking about this more, this is fine. I guess the only question is if there is value from any exception handling/logging that happens higher up the stack. Could always just change the `return;` to `throw;` if we wanted to maintain the old behaviour, if its not a `ThreadAbortException`

Reply:
If we change the `return;` to `throw;`, would there be any advantage to logging the exception since the user will be hit with a thrown exception anyway?

I think the options here could either be:
1) Only catch ThreadAbortException
2) Catch all exceptions, log them, and optimistically return without throwing

I personally don't have a preference between 1) and 2), but maybe 1) would be better so we know when something goes wrong.

Reply:
I don't know the background on this so I thought this was just an exercise to get more logging. If its also about preventing users seeing exceptions then that makes sense, but the thing to be careful of is that we don't lose any signal. Right now if users see a yellow bar, they might report a bug, so we get a signal. If we switch that to logging, then in theory we have _more_ signals because we collect everything, but if we never receive those logs, then we end up with no signal. Does that make sense?

Reply:
The original purpose of the PR was indeed just to get more logging. I see what you mean about needing to receive the logs, so I've updated the code so now we only don't throw with ThreadAbortExceptions. This preserves the original behavior of when we do/do not throw. Does that seem OK to people?

Reply:
Looks good to me!